### ISSUE TITLE
More control over test and validation set creation.

### ISSUE THREAD

**Issue by davidbaines** [sillsdev/silnlp#702]
We're often only interested in the scores for drafting from a specific source text. 
 When we have a mixed_scr configuration with data from several texts on the source side validation and test sets are created for all of them. This reduces the data for training, and may also means that the model trains is more general since the validation tests against multiple pairs.

It's possible to create multiple corpus pairs and include the test and val sets only for the drafting source of interest, however in order to do that we have to manually split the corpus books between the two corpus_pairs.

```
 corpus_pairs:
  - corpus_books: NT;GEN;EXO;RUT;JON
    mapping: mixed_src
    src:
    - eng-NIV11R
    - swhonen
    - es-RVC64
    trg: trg-
  type: train,test, val
```

It would be helpful if we could indicate test_set_src: and val_set_src:  These would be lists of the sources we want to use for the test and validation sets.

```
 corpus_pairs:
  - corpus_books: NT;GEN;EXO;RUT;JON
    mapping: mixed_src
    src:
    - eng-NIV11R
    - swhonen
    - es-RVC64
    test_set_src: eng-NIV11R
    trg: trg-trgtext
  type: train,test, val
```

When training multilingual models this would allow much more data to be used for training.

We could consider whether it should be possible to suggest sources for these that are not part of the training data.
